src/main.rs

```diff
@@ -249,13 +249,48 @@ struct CriteriaEntry {
     implies: Vec<String>,
 }
 
-/// Policies the tree must pass (TODO: understand this properly)
+/// Policies that first-party (non-foreign) crates must pass.
+///
+/// This is basically the first-party equivalent of audits.toml, which is separated out
+/// because it's not supposed to be shared (or, doesn't really make sense to share,
+/// since first-party crates are defined by "not on crates.io").
+///
+/// Because first-party crates are implicitly trusted, really the only purpose of this
+/// table is to define the boundary between a first-party crates and third-party ones.
+/// More specifically, the criteria of the dependency edges between a first-party crate
+/// and its direct third-party dependencies.
+///
+/// If this sounds overwhelming, don't worry, everything defaults to "nothing special"
+/// and an empty PolicyTable basically just means "everything should satisfy the
+/// default criteria in audits.toml".
 #[derive(serde::Serialize, serde::Deserialize)]
 struct PolicyTable {
+    /// Default criteria that must be satisfied by all *direct* third-party (foreign)
+    /// dependencies of first-party crates. If satisfied, the first-party crate is
+    /// set to satisfying all criteria.
+    ///
+    /// If not present, this defaults to the default criteria in the audits table.
```

Comment:
In the situation where the first-party crate is not a root, I think we want it to default to the parent's criteria. Only root nodes should default to the default criteria.

Example here is: I want all of geckodriver to use `safe_to_run_locally`, and just want to specify that once (at the top-level geckodriver crate) rather than duplicating it around the first-party nodes of the subtree.

Reply:
Ok so then it *does* make sense to have special rules between first-party crates, as just a way to specify that a particular sub-first-party should have such and such policy..? I'm not sure how that works if a first-party has multiple reverse-dependencies... maybe first-parties *do* need to be processed top-down?

Reply:
So basically, I'm envisioning two steps: a bottom-up traversal to compute |self| and |subtree|, and then a top-down traversal to check the policy and compute the information needed for suggest.

Reply:
policies don't "descend" into third-party transitive deps right? I don't really understand what that would mean. I guess you could take the union of your reverse-dependencies..?

Reply:
basically my confusion is that the computation of |subtree| for a first-party crate *depends* on policy, so I don't know how you handle first-party crates bottom-up? And then if you let policy descend into third-parties then I am even more confused.